bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/gems/CanopusFeature.java

```diff
@@ -128,6 +135,35 @@ public void draw(final Graphics g, final TpeImageInfo tii) {
             g2d.setColor(AO_FOV_COLOR);
             g2d.draw(s);
 
+            // Draw the windows around the primary guide stars.
+            ctx.getBaseCoordinates().forEach(bc ->
+                    CanopusWfs.Group.instance.getMembers().forEach(gp ->
+                            ctx.getTargets().getPrimaryGuideProbeTargets(gp).forEach(gpt ->
+                                    gpt.getPrimary().forEach(t ->
+                                            t.getSkycalcCoordinates(ctx.getSchedulingBlockStart()).forEach(tc -> {
+                                                // Create the transformation for this guide star's window.
+                                                final CoordinateDiff diff = new CoordinateDiff(bc, tc);
+                                                final Offset offset = diff.getOffset();
+                                                final double p = -offset.p().toArcsecs().getMagnitude();
+                                                final double q = -offset.q().toArcsecs().getMagnitude();
+
+                                                final Point2D.Double base = tii.getBaseScreenPos();
+                                                final double ppa = tii.getPixelsPerArcsec();
+
+                                                final AffineTransform tx = AffineTransform.getTranslateInstance(base.x, base.y);
+                                                tx.scale(ppa, ppa);
+                                                tx.translate(p, q);
+                                                tx.rotate(-tii.getCorrectedPosAngleRadians());
+
+                                                final Shape window = tx.createTransformedShape(flipArea(CanopusWfs.getGuideStarWindow()));
+                                                if (((ValidatableGuideProbe) gpt.getGuider()).validate(t, ctx) == GuideStarValidation.VALID)
+                                                    g2d.setColor(GS_WINDOW_COLOR);
+                                                else
+                                                    g2d.setColor(OtColor.makeSlightlyTransparent(GS_WINDOW_COLOR));
```

Comment:
`makeSlightlyTransparent` 😄

Reply:
One has to wonder what:
`OtColor.makeSlightlyTransparent(OtColor.makeSlightlyTransparent(c))`
would do. 😸

Reply:
I guess it cuts the RGB values in half, so we could get fully transparent if we continued to infinity.

Reply:
This is also where I really miss Scala `for` constructions for `Option`. All the nested `forEach` calls are kind of horrible.

Reply:
we should add `makeSlightlyLessTransparent/makeSlightlyMoreTransparent` 😄